### PR TITLE
Search page core

### PR DESCRIPTION
**Pull request by HamdanSCode**

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -86,6 +88,7 @@ dependencies {
 
     implementation 'androidx.core:core-ktx:1.9.0'
     implementation 'androidx.lifecycle:lifecycle-runtime-ktx:2.5.1'
+    implementation "androidx.compose.material3:material3:1.0.1"
```
Comment (ForzenAcademy) [inline app/build.gradle]: Api compose core?

File: app/src/main/java/com/hamdan/forzenbook/viewmodels/SearchViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+@HiltViewModel
+class SearchViewModel @Inject constructor(
+    private val postById: GetPostByUserIdUseCase,
+    private val postByQuery: GetPostByStringUseCase,
+) : BaseSearchViewModel(
+    postByQuery, postById
+) {
+
+    private val _state: MutableState<SearchState> =
+        mutableStateOf(SearchState.Content(listOf(), ""))
```
Comment (ForzenAcademy) [inline app/src/main/java/com/hamdan/forzenbook/viewmodels/SearchViewModel.kt]: Make default arguments for Content so you don't need these here

File: postCompose/src/main/java/com/hamdan/forzenbook/post/compose/Post.kt [hunk 3]
```diff
@@ -153,10 +150,9 @@ private fun ImagePostContent(
 
 @Composable
 private fun ErrorContent(
-    state: BasePostViewModel.PostState,
     onDialogDismiss: () -> Unit,
 ) {
-    StandardContent(state = state, onSendClicked = {}, onToggleClicked = {}) {
+    StandardContent(onSendClicked = {}, onToggleClicked = {}) {
```
Comment (ForzenAcademy) [inline postCompose/src/main/java/com/hamdan/forzenbook/post/compose/Post.kt]: Default arguments

**Review by ForzenAcademy — APPROVED** [review]
Some small things

File: app/build.gradle (hunk 1 above)
Comment (HamdanSCode) [reply app/build.gradle]: I could, it currently isn't bringing in uiCore and it is an application gradle file so I didn't see a reason to api the material3. Would it be better to just bring in uiCore as that is where the api'd material 3 is currently?

File: app/src/main/java/com/hamdan/forzenbook/viewmodels/SearchViewModel.kt (hunk 2 above)
Comment (HamdanSCode) [reply app/src/main/java/com/hamdan/forzenbook/viewmodels/SearchViewModel.kt]: Big true, biiiiiig true

File: app/build.gradle (hunk 1 above)
Comment (HamdanSCode) [reply app/build.gradle]: you know, I was completely mistakeing it IS in the compose core not ui core. updated.

**Pull request merged**